### DISCUSSION TITLE
Header names from scope not lowercased in datastructures.Headers constructor

### DISCUSSION THREAD

**Discussion by viccie30** [encode/starlette#2759] [Potential Issue] [answered]
`datastructures.Headers` lowercases header names if constructed from a mapping of strings https://github.com/encode/starlette/blob/ad7ff0800de563528691d26dcccf3c055cab0571/starlette/datastructures.py#L505 but not if constructed from an ASGI scope https://github.com/encode/starlette/blob/ad7ff0800de563528691d26dcccf3c055cab0571/starlette/datastructures.py#L512

According to [the ASGI specification](https://asgi.readthedocs.io/en/latest/specs/www.html) "header names should be lowercased, but it is not required".

I ran into this because I was testing some middleware by directly calling it instead of through a ASGI server. I spent more than an hour trying to figure out why `headers["Authorization"]` would throw an error when there definitely was an Authorization header in scope. I would suggest also lowercasing header names if `Headers` is constructed from a scope dictionary.

**Comment by Kludex** [comment, marked as answer]
I can see why it would be frustrating, but I can also see why the behavior is the way it is: `scope["headers"]` would be the same as passing the `raw`, which doesn't need to do the lower casing.

Also, both `TestClient` and `httpx.AsyncClient` (ASGI) do the lower casing of headers...

It's also the first time I see this complaint. I'd say the current behavior is fine.
